subprojects/core/src/integTest/groovy/org/gradle/internal/operations/notify/BuildOperationNotificationIntegrationTest.groovy

```diff
@@ -135,26 +135,21 @@ class BuildOperationNotificationIntegrationTest extends AbstractIntegrationSpec
     }
 
     // This test simulates what the build scan plugin does.
-    def "can ignore buildSrc events by deferring registration"() {
+    def "listener receives recorded operations happen before registration"() {
```

Comment:
build operation fired from composite builds could be fired early or cache configuration information are possible early fired build operations. Also new ones like the one added here: https://github.com/gradle/gradle/pull/2681 are fired early before the plugin is applied